src/Microsoft.AspNetCore.Rewrite/ConditionAbstraction/ConditionExpression.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Rewrite.Operands;
+using System;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Rewrite.ConditionAbstraction
+{
+    public class ConditionExpression : Expression
+    {
+        public Operand Operand { get; set; }
+        public bool Invert { get; set; }
+       
+        public bool CheckConditionExpression(UrlRewriteContext context, Match results, Match previous, string concatTestString)
+        {
+            // 
+            if (Operand is RegexOperand)
+            {
+                return (Operand as RegexOperand).CheckRegexOperand(previous, concatTestString);
```

Comment:
As discussed offline, refactor so you don't have to down cast. An abstract method on the base class that took an OperandContext with all of the need parameters might work. Or you could put all the state on UrlRewriteContext?